core/src/main/java/org/frankframework/pipes/IfPipe.java

```diff
@@ -0,0 +1,407 @@
+ *     <li>JSON (application/json) uses jsonPath</li>
+ * </ul>
+ * The XML mediaType is the default type, if you want to use json, you need to set this using 'mimeType' in the Message.
+ *
+ * <h4>Expressions</h4>
+ * Expressions are used to select nodes in the given input document. Imagine a collection of books:
+ * <pre>{@code
+ * {
+ *   "store": {
+ *     "book": [
+ *       {
+ *         "category": "reference",
+ *         "author": "Nigel Rees",
+ *         "title": "Sayings of the Century",
+ *         "price": 8.95
+ *       },
+ *       {
+ *         "category": "fiction",
+ *         "author": "Evelyn Waugh",
+ *         "title": "Sword of Honour",
+ *         "price": 12.99
+ *       },
+ *       {
+ *         "category": "fiction",
+ *         "author": "Herman Melville",
+ *         "title": "Moby Dick",
+ *         "isbn": "0-553-21311-3",
+ *         "price": 8.99
+ *       },
+ *       {
+ *         "category": "fiction",
+ *         "author": "J. R. R. Tolkien",
+ *         "title": "The Lord of the Rings",
+ *         "isbn": "0-395-19395-8",
+ *         "price": 22.99
+ *       }
+ *     ]
+ *   }
+ * }
+ * }</pre>
+ * <p>
+ * With both expression languages you'll be able to select one or multiple nodes from this collection.
+ * <br/>
+ * Using this pipe there are two options. Use it only with an {@code expression} or combine it with an {@code expressionValue}. When using the expression,
+ * the pipe evaluates to {@code thenForwardName} when <em>there is a match</em>, even it is empty. In the given example, this might be one of:
+ * <pre>{@code
+ *   $.store
+ *   $.store.book[1]
+ *   $.store.book[?(@.price == 22.99)].author
+ *   $.store.book[?(@.category == 'fiction')]
+ * }</pre>
+ *
+ * <h4>expressionValue</h4>
+ * When using expression combined with expressionValue, the pipe evaluates to {@code thenForwardName} when the <em>the matched value is equal to
+ * expressionValue</em>. This needs to be an exact match.
+ * <br/>
+ *
+ * <h4>XML/XPATH</h4>
+ * Xpath has been around a long time, information about the syntax can be found everywhere on the internet.
+ * The XML implementation wraps the Xpath expression in an XSL. This enables us to use complex expressions which evaluate to true or false instead
+ * being used only as a selector of nodes in the input XML. This is available to be backwards compatible with the {@link XmlIf} pipe.
+ * For instance, take the following example input:
+ * <pre>{@code
+ *   <results>
+ *     <result name="test"></result>
+ *     <result name="test"></result>
+ *   </results>
+ * }</pre>
+ * Examples with complex expressions might be something like: {@code number(count(/results/result[contains(@name , 'test')])) > 1}, to test if there's more
+ * than one node found containing the string 'test'. Please check if a simpler, less error prone expression like
+ * {@code /results/result[contains(@name, 'test')]} can suffice.
+ * <p></p>
+ *
+ * <h4>Changes compared to the XmlIf pipe</h4>
```

Comment:
@nielsm5 , this might be a problem for the 'xmlif extends ifpipe' idea we talked about.